app/services/activitypub/fetch_replies_service.rb

```diff
@@ -54,10 +81,19 @@ def fetch_collection(collection_or_uri)
   end
 
   def filtered_replies
-    # Only fetch replies to the same server as the original status to avoid
-    # amplification attacks.
+    if @all_replies
+      # Reject all statuses that we already have in the db
+      @items.map { |item| value_or_id(item) }.reject { |uri| Status.exists?(uri: uri) }
```

Comment:
N+1 queries, this could potentially be done in a single query creating a map then checking if the item is in the map, see `Account::Interactions` module for an example.

Reply:
tried to address this by making arel scopes so that we can do batch pluck/touch operations as needed